org-code-javabuilder/theater/src/main/java/org/code/theater/Theater.java

```diff
@@ -26,7 +27,12 @@ private static void playInternal(Scene... scenes) {
     for (Scene scene : scenes) {
       allActions.addAll(scene.getActions());
     }
-
-    TheaterPlayer.getInstance().play(allActions);
+    JavabuilderContext context = JavabuilderContext.getInstance();
+    if (!context.containsKey(TheaterPlayer.class)) {
+      context.register(TheaterPlayer.class, new TheaterPlayer());
+    }
+    TheaterPlayer theaterPlayer =
+        (TheaterPlayer) JavabuilderContext.getInstance().get(TheaterPlayer.class);
+    theaterPlayer.play(allActions);
```

Comment:
Just an idea, and this can certainly be a follow-up if we want to do this - but this got me thinking about if we need mini-app specific initialization code that runs on every execution, similar to our overall initialization code in our handlers. Specifically, mini-apps could use this to register any singletons they want with `JavabuilderContext` so it all happens in one place before execution starts.

Reply:
Would we decide to run this initialization code based on some flag that is sent to Javabuilder? Right now a given instance of javabuilder can run any mini app, and I don't think we actually send the mini app type when we run code (although we certainly could).

Reply:
Yeah, not sure exactly - maybe we initialize the mini-app based on the lambda? But you're right, any instance can run any mini-app, so maybe we would just initialize all mini-app resources? I think it would depend on what the initialization ends up becoming. If it's just registering static objects then it doesn't seem too bad for all mini-apps to register their static objects on startup but if it's something like memory allocation then we'd want to probably only initialize when required. This was more of a code organization idea, so I can see it going both ways.